src/http/server.clj

```diff
@@ -325,6 +334,34 @@
          (log/infof "\"%s %s HTTP/1.1\" %s %s" method'  uri status len))
        response))))
 
+(defn password-prompt! []
+  (if-let [console (System/console)]
+    (let [pw (String. (.readPassword console  "Enter password: " nil))]
+      (if (empty? pw)
+        (do (println "Password cannot be empty!")
+            (recur))
+        pw))
+    (throw (Exception. "No console available"))))
+
+(defn maybe-inject-auth
+  [auth handler]
+  (if (some? auth)
+    (fn [{:keys [headers] :as req}]
+      (if (not= (headers "Authorization") auth)
+        (handler (-> (update req :headers dissoc "authorization")
+                     (assoc-in [:headers "WWW-Authenticate"] "Basic")
+                     (assoc :status 401)))
+        (handler req)))
+    handler))
+
+(defn parse-auth [auth]
+  "make sure password is present, if not prompt for it"
```

Comment:
This is not a docstring, docstring should go after function name before params definition.